`solana-address-lookup-table-program` compile error in program environment

#### Problem

`solana-address-lookup-table-program ^1.17` is broken in programs after https://github.com/solana-labs/solana/pull/33165.

It uses `solana_sdk` crate instead of `solana_program` which cause compile errors for programs:

https://github.com/solana-labs/solana/blob/6f2b3bbad38f8360efb7272324d487cfd4b9f48e/programs/address-lookup-table/src/lib.rs#L7-L15

```
   Compiling solana-address-lookup-table-program v1.17.0
error[E0433]: failed to resolve: use of undeclared crate or module `solana_sdk`
  --> src/lib.rs:11:9
   |
11 | pub use solana_sdk::address_lookup_table::{
   |         ^^^^^^^^^^ use of undeclared crate or module `solana_sdk`
```

To be specific, the above error is about `solana-sdk` is not being declared as a dependency in program environment but it would still fail if it was declared:

https://github.com/solana-labs/solana/blob/6f2b3bbad38f8360efb7272324d487cfd4b9f48e/programs/address-lookup-table/Cargo.toml#L24-L26

#### Proposed Solution

Export it from `solana-program` instead of `solana-sdk` for program compilation.

--- a/programs/address-lookup-table/src/lib.rs
+++ b/programs/address-lookup-table/src/lib.rs
@@ -5,18 +5,6 @@
 #[cfg(not(target_os = "solana"))]
 pub mod processor;
 
-#[cfg(not(target_os = "solana"))]
-#[deprecated(
-    since = "1.17.0",
-    note = "Please use `solana_sdk::address_lookup_table` instead"
-)]
-pub use solana_sdk::address_lookup_table::{
-    error, instruction,
-    program::{check_id, id, ID},
-    state,
-};
-
-#[cfg(target_os = "solana")]
 #[deprecated(
     since = "1.17.0",
     note = "Please use `solana_program::address_lookup_table` instead"

--- a/programs/address-lookup-table/src/lib.rs
+++ b/programs/address-lookup-table/src/lib.rs
@@ -2,6 +2,7 @@
 #![cfg_attr(RUSTC_WITH_SPECIALIZATION, feature(specialization))]
 #![cfg_attr(RUSTC_NEEDS_PROC_MACRO_HYGIENE, feature(proc_macro_hygiene))]
 
+#[cfg(not(target_os = "solana"))]
 pub mod processor;
 
 #[cfg(not(target_os = "solana"))]

--- a/programs/address-lookup-table/src/lib.rs
+++ b/programs/address-lookup-table/src/lib.rs
@@ -4,11 +4,23 @@
 
 pub mod processor;
 
+#[cfg(not(target_os = "solana"))]
 #[deprecated(
     since = "1.17.0",
     note = "Please use `solana_sdk::address_lookup_table` instead"
 )]
 pub use solana_sdk::address_lookup_table::{
+    error, instruction,
+    program::{check_id, id, ID},
+    state,
+};
+
+#[cfg(target_os = "solana")]
+#[deprecated(
+    since = "1.17.0",
+    note = "Please use `solana_program::address_lookup_table` instead"
+)]
+pub use solana_program::address_lookup_table::{
     error, instruction,
     program::{check_id, id, ID},
     state,

